utils_linux.go

```diff
@@ -17,3 +25,33 @@ func (p *Pinger) matchID(ID int) bool {
 	}
 	return true
 }
+
+func getConnFD(conn *icmp.PacketConn) (fd int) {
+	var packetConn reflect.Value
+
+	defer func() {
+		if r := recover(); r != nil {
+			fd = -1
+		}
+	}()
+
+	if conn.IPv4PacketConn() != nil {
+		packetConn = reflect.ValueOf(conn.IPv4PacketConn().PacketConn)
+	} else if conn.IPv6PacketConn() != nil {
+		packetConn = reflect.ValueOf(conn.IPv6PacketConn().PacketConn)
+	} else {
+		return -1
+	}
+
+	netFD := reflect.Indirect(reflect.Indirect(packetConn).FieldByName("fd"))
+	pollFD := netFD.FieldByName("pfd")
+	systemFD := pollFD.FieldByName("Sysfd")
```

Comment:
/Ouch... this is poking around _internals_ of multiple packages, so it's very brittle.

I suppose the reason why the FD is not exposed is because that's a very Unix-centric concept.

Did you try going the other way around: propose the addition of BindToDevice in `icmp.PacketConn`?